src/cpp/cpp_typecheck_conversions.cpp

```diff
@@ -220,24 +219,21 @@ bool cpp_typecheckt::standard_conversion_integral_promotion(
     std::size_t width=to_unsignedbv_type(expr.type()).get_width();
     if(width >= config.ansi_c.int_width)
       return false;
-    new_expr=expr;
     if(width==config.ansi_c.int_width)
       int_type.id(ID_unsignedbv);
-    new_expr.make_typecast(int_type);
+    new_expr = typecast_exprt::conditional_cast(expr, int_type);
     return true;
```

Comment:
Same here -- the cast will always happen.

Reply:
I don't think so - the line above the change would make the typecast a no-op.

Reply:
Unrelated to PR: the two lines above the change are dead code (note the branch directly before). There's probably a bug there.

Reply:
Oh, indeed. Aren't those two lines really just unnecessary, and we should do an unconditional cast as you suggested? To me it seems there no bug other than it being dead code.

Reply:
The bug is that an "unsigned int" will be promoted to "signed int", which is wrong.

Reply:
I don't think so, the `width >= config.ansi_c.int_width` would catch that?